app/src/main/java/org/systers/mentorship/view/activities/MainActivity.kt

```diff
@@ -42,33 +44,38 @@ class MainActivity : BaseActivity() {
             BottomNavigationView.OnNavigationItemSelectedListener { item ->
                 when (item.itemId) {
                     R.id.navigation_home -> {
-                        replaceFragment(R.id.contentFrame, HomeFragment.newInstance(),
+                        replaceFragment(R.id.contentFrame, activeFragment, HomeFragment.newInstance(),
                                 R.string.fragment_title_home)
                         atHome = true
+                        activeFragment = HomeFragment()
```

Comment:
What you are doing here is creating a new instance of `HomeFragment` which is wrong. Same you are doing for other fragments. This defeats the purpose of creating the `newInstance()` method that we have created in every fragment.

You are using `activeFragment` only for the name of the fragment, which can be done using a string.

Reply:
Yea, understandabe. Will update here and refactor to pass strings instead creating instances.